src/App.jsx

```diff
@@ -14,45 +17,103 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_BY_INITIAL = '';
+const SORT_BY_ALPHABET = 'name';
+const SORT_BY_LENGTH = 'length';
+
+function getPreparedGoods(goods, sortedBy, isReversed) {
+  const preparedGoods = goods.map(good => (
+    {
+      id: goods.indexOf(good),
+      name: good,
+      length: good.length,
+    }
+  ));
+
+  if (sortedBy) {
+    preparedGoods.sort((goodFirst, goodSecond) => {
+      switch (sortedBy) {
+        case SORT_BY_ALPHABET:
+          return goodFirst[sortedBy].localeCompare(goodSecond[sortedBy]);
+
+        case SORT_BY_LENGTH:
+          return goodFirst[sortedBy] - goodSecond[sortedBy];
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [sortedBy, setSortedBy] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, sortedBy, isReversed);
+
+  const reset = () => {
+    setSortedBy(SORT_BY_INITIAL);
+    setIsReversed(false);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn(
+            'button is-info',
+            { 'is-light': sortedBy !== SORT_BY_ALPHABET },
+          )}
+          onClick={() => setSortedBy(SORT_BY_ALPHABET)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn(
+            'button is-success',
+            { 'is-light': sortedBy !== SORT_BY_LENGTH },
+          )}
+          onClick={() => setSortedBy(SORT_BY_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn(
+            'button is-warning',
+            { 'is-light': isReversed !== true },
```

Comment:
maybe you forgot to change the comparison (isReversed !== true). In below code you added better option (!isReversed)

Reply:
➕ , boolean values can be simplified